### ISSUE TITLE
Erroneous end coordinates

### ISSUE THREAD

**Issue by handstad** [mdshw5/pyfaidx#79]
Using pyfaidx 0.4.5.2, the end coordinate attribute is wrong:

```python
from pyfaidx import Fasta
f = Fasta("/path/to/human_g1k_v37_decoy.fasta", one_based_attributes=False)
s = f[1][10000:10005]
src = -s

s.end # Returns 5 (not 10005)
src.start #  Returns 5
```

**Comment by mdshw5** [comment]
Bummer. I guess this is another test to add. I'm sure that the issue is `one_based_attributes`, which I probably forgot to make a case for. I'll fix this with a point release.

**Comment by mdshw5** [comment]
Fixed this in 5623c74. @handstad: if you could test on your use cases and report success/failure I would appreciate the feedback. I'm going to push these changes as a point release since all tests, including a few new ones I added, are passing.

**Comment by mdshw5** [comment]
Looks like CI reports failure so I must have forgotten something.

**Comment by huguesfontenelle** [comment]
`s.end`
returns the length of the sequence, not the genomic coordinate (I'd expect `10005` here, for this test case).

**Comment by mdshw5** [comment]
Yes, this is still a known issue. Thanks for testing. I attempted a fix that turned out to be incomplete so I'm putting in the time to make sure I don't have to fix this yet again. This time of year is just very busy for me so I appreciate your patience.

**Comment by mdshw5** [comment]
Fixed in 0.4.7.1
